Bug: https://github.com/rust-lang/cargo/issues/10005

Index: rust/src/tools/cargo/tests/testsuite/cfg.rs
===================================================================
--- rust.orig/src/tools/cargo/tests/testsuite/cfg.rs
+++ rust/src/tools/cargo/tests/testsuite/cfg.rs
@@ -272,7 +272,7 @@ fn any_ok() {
 
 // https://github.com/rust-lang/cargo/issues/5313
 #[cargo_test]
-#[cfg(all(target_arch = "x86_64", target_os = "linux", target_env = "gnu"))]
+#[cfg(all(target_arch = "x86_64", target_os = "linux", target_env = "gnu", target_pointer_width = "64"))]
 fn cfg_looks_at_rustflags_for_target() {
     let p = project()
         .file(
