Index: rust/src/tools/cargo/Cargo.toml
===================================================================
--- rust.orig/src/tools/cargo/Cargo.toml
+++ rust/src/tools/cargo/Cargo.toml
@@ -33,8 +33,8 @@ env_logger = "0.10.0"
 filetime = "0.2.9"
 flate2 = { version = "1.0.3", default-features = false, features = ["zlib"] }
 fwdansi = "1.1.0"
-git2 = "0.17.1"
-git2-curl = "0.18.0"
+git2 = "0.18"
+git2-curl = "0.19.0"
 gix = { version = "0.44.1", default-features = false, features = ["blocking-http-transport-curl", "progress-tree"] }
 gix-features-for-configuration-only = { version = "0.29.0", package = "gix-features", features = [ "parallel" ] }
 glob = "0.3.0"
@@ -53,7 +53,7 @@ jobserver = "0.1.26"
 lazy_static = "1.3.0"
 lazycell = "1.2.0"
 libc = "0.2.144"
-libgit2-sys = "0.15.1"
+libgit2-sys = "0.16.1"
 log = "0.4.17"
 memchr = "2.1.3"
 miow = "0.5.0"
Index: rust/src/tools/cargo/crates/cargo-test-support/Cargo.toml
===================================================================
--- rust.orig/src/tools/cargo/crates/cargo-test-support/Cargo.toml
+++ rust/src/tools/cargo/crates/cargo-test-support/Cargo.toml
@@ -15,7 +15,7 @@ cargo-util = { path = "../cargo-util" }
 crates-io = { path = "../crates-io" }
 filetime = "0.2"
 flate2 = { version = "1.0", default-features = false, features = ["zlib"] }
-git2 = "0.17.0"
+git2 = "0.18.0"
 glob = "0.3"
 itertools = "0.10.0"
 lazy_static = "1.0"
