From: Debian Rust Maintainers <pkg-rust-maintainers@alioth-lists.debian.net>
Date: Mon, 6 May 2024 10:25:32 +0200
Subject: d-0011-cargo-remove-nghttp2

Last-Update: 2023-05-17 <YYYY-MM-DD, last update of the meta-information, optional>
---
 vendor/curl-sys/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vendor/curl-sys/Cargo.toml b/vendor/curl-sys/Cargo.toml
index 945ee8e..04b9800 100644
--- a/vendor/curl-sys/Cargo.toml
+++ b/vendor/curl-sys/Cargo.toml
@@ -52,7 +52,7 @@ version = "0.3.3"
 [features]
 default = ["ssl"]
 force-system-lib-on-osx = []
-http2 = ["libnghttp2-sys"]
+http2 = []
 mesalink = []
 ntlm = []
 poll_7_68_0 = []
