From: Debian Rust Maintainers <pkg-rust-maintainers@alioth-lists.debian.net>
Date: Mon, 6 May 2024 10:25:32 +0200
Subject: d-0010-cargo-remove-vendored-c-crates

Last-Update: 2023-05-17
---
 src/tools/cargo/Cargo.toml | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/tools/cargo/Cargo.toml b/src/tools/cargo/Cargo.toml
index 3fb36b4..c011909 100644
--- a/src/tools/cargo/Cargo.toml
+++ b/src/tools/cargo/Cargo.toml
@@ -226,7 +226,5 @@ test = false
 doc = false
 
 [features]
-vendored-openssl = ["openssl/vendored"]
-vendored-libgit2 = ["libgit2-sys/vendored"]
+# Debian: removed vendoring flags
 # This is primarily used by rust-lang/rust distributing cargo the executable.
-all-static = ['vendored-openssl', 'curl/static-curl', 'curl/force-system-lib-on-osx']
