From: Debian Rust Maintainers <pkg-rust-maintainers@alioth-lists.debian.net>
Date: Sat, 2 Oct 2021 01:07:59 +0100
Subject: d-0001-disable-miniz

---
 vendor/flate2/Cargo.toml | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/vendor/flate2/Cargo.toml b/vendor/flate2/Cargo.toml
index 616f6ee..a593593 100644
--- a/vendor/flate2/Cargo.toml
+++ b/vendor/flate2/Cargo.toml
@@ -40,10 +40,6 @@ version = "1.1.0"
 optional = true
 default-features = false
 
-[dependencies.miniz-sys]
-version = "0.1.11"
-optional = true
-
 [dependencies.miniz_oxide]
 version = "0.5.0"
 optional = true
