partial revert of b9eedea4b0368fd1f00f204db75109ff444fab5b upstream

--- a/src/bootstrap/dist.rs
+++ b/src/bootstrap/dist.rs
@@ -72,6 +72,7 @@
         tarball.set_product_name("Rust Documentation");
         tarball.add_bulk_dir(&builder.doc_out(host), dest);
         tarball.add_file(&builder.src.join("src/doc/robots.txt"), dest, 0o644);
+        tarball.permit_symlinks(true);
         Some(tarball.generate())
     }
 }
