--- a/vendor/rustix/src/termios/constants.rs
+++ b/vendor/rustix/src/termios/constants.rs
@@ -33,6 +33,8 @@
 )))]
 pub use backend::termios::types::B1500000;
 #[cfg(not(any(
+    target_arch = "sparc",
+    target_arch = "sparc64",
     target_os = "dragonfly",
     target_os = "freebsd",
     target_os = "haiku",
@@ -96,6 +98,8 @@
 )))]
 pub use backend::termios::types::B4000000;
 #[cfg(not(any(
+    target_arch = "sparc",
+    target_arch = "sparc64",
     target_os = "dragonfly",
     target_os = "haiku",
     target_os = "ios",
@@ -104,6 +108,8 @@
 )))]
 pub use backend::termios::types::B460800;
 #[cfg(not(any(
+    target_arch = "sparc",
+    target_arch = "sparc64",
     target_os = "dragonfly",
     target_os = "freebsd",
     target_os = "haiku",
@@ -116,6 +122,8 @@
 )))]
 pub use backend::termios::types::B500000;
 #[cfg(not(any(
+    target_arch = "sparc",
+    target_arch = "sparc64",
     target_os = "dragonfly",
     target_os = "freebsd",
     target_os = "haiku",
@@ -732,6 +740,8 @@
         )))]
         backend::termios::types::B500000 => Some(500_000),
         #[cfg(not(any(
+            target_arch = "sparc",
+            target_arch = "sparc64",
             target_os = "dragonfly",
             target_os = "freebsd",
             target_os = "haiku",
@@ -744,6 +754,8 @@
         )))]
         backend::termios::types::B576000 => Some(576_000),
         #[cfg(not(any(
+            target_arch = "sparc",
+            target_arch = "sparc64",
             target_os = "dragonfly",
             target_os = "haiku",
             target_os = "ios",
@@ -752,6 +764,8 @@
         )))]
         backend::termios::types::B921600 => Some(921_600),
         #[cfg(not(any(
+            target_arch = "sparc",
+            target_arch = "sparc64",
             target_os = "dragonfly",
             target_os = "freebsd",
             target_os = "haiku",
--- a/vendor/rustix/src/termios/mod.rs
+++ b/vendor/rustix/src/termios/mod.rs
@@ -41,6 +41,8 @@
 )))]
 pub use constants::B1500000;
 #[cfg(not(any(
+    target_arch = "sparc",
+    target_arch = "sparc64",
     target_os = "dragonfly",
     target_os = "freebsd",
     target_os = "haiku",
@@ -104,6 +106,8 @@
 )))]
 pub use constants::B4000000;
 #[cfg(not(any(
+    target_arch = "sparc",
+    target_arch = "sparc64",
     target_os = "dragonfly",
     target_os = "haiku",
     target_os = "ios",
@@ -112,6 +116,8 @@
 )))]
 pub use constants::B460800;
 #[cfg(not(any(
+    target_arch = "sparc",
+    target_arch = "sparc64",
     target_os = "dragonfly",
     target_os = "freebsd",
     target_os = "haiku",
@@ -124,6 +130,8 @@
 )))]
 pub use constants::B500000;
 #[cfg(not(any(
+    target_arch = "sparc",
+    target_arch = "sparc64",
     target_os = "dragonfly",
     target_os = "freebsd",
     target_os = "haiku",
