--- a/compiler/rustc_lint_defs/src/builtin.rs
+++ b/compiler/rustc_lint_defs/src/builtin.rs
@@ -2883,11 +2883,13 @@
     ///
     /// use std::arch::asm;
     ///
+    /// #[cfg(not(any(target_arch = "powerpc64", target_arch = "s390x")))]
     /// #[naked]
     /// pub fn default_abi() -> u32 {
     ///     unsafe { asm!("", options(noreturn)); }
     /// }
     ///
+    /// #[cfg(not(any(target_arch = "powerpc64", target_arch = "s390x")))]
     /// #[naked]
     /// pub extern "Rust" fn rust_abi() -> u32 {
     ///     unsafe { asm!("", options(noreturn)); }
--- a/compiler/rustc_lint/src/builtin.rs
+++ b/compiler/rustc_lint/src/builtin.rs
@@ -3130,7 +3130,10 @@
     /// ### Example
     ///
     /// ```rust,compile_fail
-    /// # #![feature(asm_experimental_arch)]
+    /// #![cfg_attr(
+    ///  not(any(target_arch = "powerpc64", target_arch = "s390x")),
+    ///  feature(asm_experimental_arch)
+    /// )]
     /// use std::arch::asm;
     ///
     /// fn main() {
